src/ledger/NetworkConfig.cpp

```diff
@@ -873,7 +881,9 @@ updateMemCostParamsEntryForV22(AbstractLedgerTxn& ltxRoot)
                        .contractCostParamsMemBytes();
 
     auto const& vals = xdr::xdr_traits<ContractCostType>::enum_values();
-    params.resize(static_cast<uint32>(vals.size()));
+
+    // Resize to fit the last cost type added in v22
+    params.resize(static_cast<uint32>(ContractCostType::Bls12381FrInv) + 1);
```

Comment:
We should static assert on the size here to make sure there isn't a mistake in the enum numbering.

Reply:
Nvm this isn't possible. but I verified that the numbers are right.